src/client/clientevent.h

```diff
@@ -66,11 +67,14 @@ struct ClientEventHudChange
 
 struct ClientEvent
 {
+	// TODO: should get rid of this ctor
+	ClientEvent() : type(CE_NONE) {}
+
+	ClientEvent(ClientEventType type) : type(type) {}
```

Comment:
Maybe leave it the old way and use
```
	ClientEvent *event = new ClientEvent();
	event->type = ...;
```
And then improve it wholesale in a separate PR.

Reply:
Sure, it's work half-done. But it provides a good indication what the idea was to replace the default ctor with.